### ISSUE TITLE
Clients build failed when set --build-arg "PYVER=3.6"

### ISSUE THREAD

**Issue by luuil** [NVIDIA/tensorrt-inference-server#28]
Will failed

```bash
sudo docker build -t tensorrtserver_clients --target trtserver_build --build-arg "PYVER=3.6" --build-arg "BUILD_CLIENTS_ONLY=1" .
```
but 3.5 works

```bash
sudo docker build -t tensorrtserver_clients --target trtserver_build --build-arg "PYVER=3.5" --build-arg "BUILD_CLIENTS_ONLY=1" .
```

**Comment by deadeyegoodwin** [comment]
We only test Python 2.7 and 3.5. Can you include the output showing exactly what failed.

**Comment by mrjackbo** [comment]
I have also had problems building the TRTIS client wheel for targets other than ubuntu 16.04 with Python 3.5. With the old github repository this was easily possible, but now there are dependencies in the Dockerfile which make this more complicated.
 - The problem is in the build step using the nvidia/tensorflow image. This is an image based on ubuntu 16.04 with python 3.5 
 - /opt/tensorflow/nvbuild.sh is executed in nvidia/tensorflow,
 - One can work around usage of nvbuild.sh and create a wheel for a different python version, e.g. by installing miniconda with some other python version inside nvidia/tensorflow. But I was not able to build a wheel usable under ubuntu 18.04. The problem is that libcurl changed from 16.04 to 18.04, so to make a version of the client wheel usable under 18.04, one would need a version of the nvidia tensorflow image, based on ubuntu 18.04

In my opinion (and this is a feature request), the build process for the clients should be separated from the build process of the server, and made much simpler. After all (correct me if I'm wrong), the clients do not depend on tensorflow, tensorflow-serving, pytorch, caffe2 etc. One should only need grpc, libcurl, and the protobuf service definitions and hence the clients should be compilable in an almost generic current linux image. The compilation should also be quite quick...

**Comment by luuil** [comment]
@deadeyegoodwin Hi, in my case, output details as below
```bash
// Omitt irrelevant information above..
Step 41/46 : RUN (cd /opt/tensorflow && ./nvbuild.sh --python$PYVER --configonly) &&     (cd tools && mv bazel.rc bazel.orig &&      cat bazel.orig /opt/tensorflow/.tf_configure.bazelrc > bazel.rc) &&     bash -c 'if [ "$BUILD_CLIENTS_ONLY" != "1" ]; then                bazel build -c opt --config=cuda src/servers/trtserver src/clients/... src/test/...;              else                bazel build -c opt src/clients/...;              fi' &&     (cd /opt/tensorrtserver && ln -s /workspace/qa qa) &&     mkdir -p /opt/tensorrtserver/bin &&     cp bazel-bin/src/clients/c++/image_client /opt/tensorrtserver/bin/. &&     cp bazel-bin/src/clients/c++/perf_client /opt/tensorrtserver/bin/. &&     cp bazel-bin/src/clients/c++/simple_client /opt/tensorrtserver/bin/. &&     mkdir -p /opt/tensorrtserver/lib &&     cp bazel-bin/src/clients/c++/librequest.so /opt/tensorrtserver/lib/. &&     cp bazel-bin/src/clients/c++/librequest.a /opt/tensorrtserver/lib/. &&     mkdir -p /opt/tensorrtserver/pip &&     bazel-bin/src/clients/python/build_pip /opt/tensorrtserver/pip/. &&     bash -c 'if [ "$BUILD_CLIENTS_ONLY" != "1" ]; then                cp bazel-bin/src/servers/trtserver /opt/tensorrtserver/bin/.;                cp bazel-bin/src/test/caffe2plan /opt/tensorrtserver/bin/.;              fi' &&     bazel clean --expunge &&     rm -rf /root/.cache/bazel &&     rm -rf /tmp/*
 ---> Running in aade39c88213
UNKNOWN OPTION --python3.6
Run ./nvbuild.sh -h for help
The command '/bin/sh -c (cd /opt/tensorflow && ./nvbuild.sh --python$PYVER --configonly) &&     (cd tools && mv bazel.rc bazel.orig &&      cat bazel.orig /opt/tensorflow/.tf_configure.bazelrc > bazel.rc) &&     bash -c 'if [ "$BUILD_CLIENTS_ONLY" != "1" ]; then                bazel build -c opt --config=cuda src/servers/trtserver src/clients/... src/test/...;              else                bazel build -c opt src/clients/...;              fi' &&     (cd /opt/tensorrtserver && ln -s /workspace/qa qa) &&     mkdir -p /opt/tensorrtserver/bin &&     cp bazel-bin/src/clients/c++/image_client /opt/tensorrtserver/bin/. &&     cp bazel-bin/src/clients/c++/perf_client /opt/tensorrtserver/bin/. &&     cp bazel-bin/src/clients/c++/simple_client /opt/tensorrtserver/bin/. &&     mkdir -p /opt/tensorrtserver/lib &&     cp bazel-bin/src/clients/c++/librequest.so /opt/tensorrtserver/lib/. &&     cp bazel-bin/src/clients/c++/librequest.a /opt/tensorrtserver/lib/. &&     mkdir -p /opt/tensorrtserver/pip &&     bazel-bin/src/clients/python/build_pip /opt/tensorrtserver/pip/. &&     bash -c 'if [ "$BUILD_CLIENTS_ONLY" != "1" ]; then                cp bazel-bin/src/servers/trtserver /opt/tensorrtserver/bin/.;                cp bazel-bin/src/test/caffe2plan /opt/tensorrtserver/bin/.;              fi' &&     bazel clean --expunge &&     rm -rf /root/.cache/bazel &&     rm -rf /tmp/*' returned a non-zero code: 1
```

**Comment by deadeyegoodwin** [comment]
@luuil Thanks for the output, we will look into fixing that issue.

@mrjackbo I am surprised that a BUILD_CLIENTS_ONLY=1 build takes several hours. That said, we will look into improving the client-only build to be easier to use and faster.

**Comment by mrjackbo** [comment]
Thanks for your reply. The phrase "several hours" is not mine, and in my experience a BUILD_CLIENTS_ONLY=1 build is reasonably quick...one does have to download several GB of dependencies though.

**Comment by elmuz** [comment]
@mrjackbo I was able to create a whl for Python 3.6 by editing nvbuild.sh file as you mentioned. Then I could install it by using pip3. Next, if you install [libucurl3](https://packages.ubuntu.com/bionic/libcurl3) you are able to proper load the client library (i.e. in a python `from tensorrtserver.api import *` doesn't complain).
I didn't test the client properly though.

**Comment by mrjackbo** [comment]
@elmuz I have also been able to build such wheels. Perhaps the libcurl3 package in Ubuntu was updated recently? I am sure that the ubuntu18 Version of libcurl3 did not work previously...

**Comment by elmuz** [comment]
Today I tested a minimal client setup with the _.whl_ for Python 3.6 and _libcurl3_ from official repository of Ubuntu 18.04. It works fine.
The problem comes as soon as you want to install a package (_cmake_ in my case) which relies on _libcurl4_. Unfortunately installing libcurl4 replaces the older version...

**Comment by deadeyegoodwin** [comment]
With https://github.com/NVIDIA/tensorrt-inference-server/pull/132 we now have a separate Makefile.client (and Dockerfile.client for convenience) that is used to build the client libraries and examples. The main Dockerfile now builds just the server.

Give it a try with your target OS and Python version and send in pull requests to fix any issues you find.

**Comment by rohitrawat** [comment]
Since Python3.6 is not available in Ubuntu 16.04 apt repos, I added the following to Dockerfile.client to make it work:
```
32,34d31
< RUN apt-get update && apt-get install -y software-properties-common
< RUN add-apt-repository -y ppa:deadsnakes/ppa
< 
```

**Comment by deadeyegoodwin** [comment]
@rohitrawat Thanks for the info.
BTW, we are planning on updating the client build to support python 3.6 for the 19.05 release, and to include pre-build client libraries for both ubuntu 16.04 and 18.04.
